tests/unit/plugins/modules/conftest.py

```diff
@@ -5,45 +5,44 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+import contextlib as _contextlib
 import json
 
 import pytest
 
 from ansible.module_utils.six import string_types
-from ansible.module_utils.common.text.converters import to_bytes
 from ansible.module_utils.common._collections_compat import MutableMapping
 
 from ansible_collections.community.general.plugins.module_utils import deps
+from ansible_collections.community.general.tests.unit.plugins.modules.utils import set_module_args as _set_module_args
 
 
-def fix_ansible_args(args):
+def _fix_ansible_args(args):
     if isinstance(args, string_types):
-        return args
+        # This should be deprecated!
+        return json.loads(args)
 
     if isinstance(args, MutableMapping):
-        if 'ANSIBLE_MODULE_ARGS' not in args:
-            args = {'ANSIBLE_MODULE_ARGS': args}
-        if '_ansible_remote_tmp' not in args['ANSIBLE_MODULE_ARGS']:
-            args['ANSIBLE_MODULE_ARGS']['_ansible_remote_tmp'] = '/tmp'
-        if '_ansible_keep_remote_files' not in args['ANSIBLE_MODULE_ARGS']:
-            args['ANSIBLE_MODULE_ARGS']['_ansible_keep_remote_files'] = False
-        args = json.dumps(args)
         return args
 
-    else:
-        raise Exception('Malformed data to the patch_ansible_module pytest fixture')
+    raise Exception('Malformed data to the patch_ansible_module pytest fixture')
+
+
+@pytest.fixture
+def patch_ansible_module(request):
+    args = _fix_ansible_args(request.param)
+    with _set_module_args(args):
+        yield
 
 
 @pytest.fixture
-def patch_ansible_module(request, mocker):
-    if hasattr(request, "param"):
-        args = fix_ansible_args(request.param)
-        mocker.patch('ansible.module_utils.basic._ANSIBLE_ARGS', to_bytes(args))
-    else:
-        def _patch(args):
-            args = fix_ansible_args(args)
-            mocker.patch('ansible.module_utils.basic._ANSIBLE_ARGS', to_bytes(args))
-        return _patch
+def patch_ansible_module_uthelper(request):
+    @_contextlib.contextmanager
```

Comment:
Why the distinction?

Reply:
Because you cannot mix context managers with regular functions.

Reply:
Longer text: once you have a `yield` in a function (which you need for using the `_set_module_args` context manager) you cannot do the previous `if`/`else` and simply return a function in the second case. You'd need to use `yield` there too instead of `return`. But then it's cleaner/easier to simply split this up into two different fixtures, since the behavior is very different between the two.